packages/@aws-cdk/aws-eks-legacy/lib/kubectl-layer.ts

```diff
@@ -68,8 +68,4 @@ export class KubectlLayer extends Resource implements lambda.ILayerVersion {
 
     this.layerVersionArn = Token.asString(resource.getAtt('Outputs.LayerVersionArn'));
   }
-
-  public addPermission(_id: string, _permission: lambda.LayerVersionPermission): void {
-    return;
-  }
```

Comment:
This removal seems like a rather large change. Why did `KubectlLayer` not support adding permissions and how come that it does support that now?

Reply:
FWIW the only issue here is that changing the return type would not work for this scenario. I also considered having this function throw an error indicating that permissions were not supported.